### PR TITLE
JIRA ticket regex update

### PR DESCRIPTION
**Pull request by ashfurrow**
This work is based on the RFC here:https://github.com/artsy/README/issues/274

### CODE REVIEW

**Comment by ashfurrow** [comment]
Peril failures will be fixed once this commit is merged: https://github.com/artsy/peril-settings/pull/144/commits/aa2fb47a7e39f4f768a65519d69188d4d3101c87

File: org/jira/utils.ts [hunk 1]
```diff
@@ -1,29 +1,24 @@
-const companyPrefix = "artsyproduct"
+import { flatten } from "lodash"
 
 // https://stackoverflow.com/questions/19322669/regular-expression-for-a-jira-identifier#30518972
-const jiraTicketRegex = /\d+-[A-Z]+(?!-?[a-zA-Z]{1,10})/g
-const urlSuffix = `.atlassian.net\/browse\/\d+-[A-Z]+(?!-?[a-zA-Z]{1,10})/g`
-const jiraURLRegex = new RegExp("https://" + companyPrefix + urlSuffix)
+const jiraTicketRegex = /[\]\)](?<ticketID>\d+-[A-Z]+(?!-?[a-zA-Z]{1,10}))[\[\(]/g
```
Comment (anandaroop) [inline org/jira/utils.ts]: Nice use of named capture groups 🤩

Comment (ashfurrow) [reply org/jira/utils.ts]: Thanks! It was @dblandin's idea 👍

**Comment by ashfurrow** [comment]
I have added https://github.com/artsy/peril-settings/pull/144/commits/e09f7afde6bc0f9f5f235a9c9b571fead50544bc to handle the case of Markdown links (eg: `[ABC-123](https://artsyprodu...)`) that @erikdstock pointed out here: https://github.com/artsy/README/issues/274#issuecomment-565164705

**Comment by dblandin** [comment]
Going to merge this in! :tada:

**Pull request merged**